### ISSUE TITLE
Launch KetherNFT: The NFT wrapper for Kether ads

### ISSUE THREAD

**Issue by shazow** [thousandetherhomepage/ketherhomepage#48]
Meta-issue for getting the KetherNFT contract production-ready.

- [x] [KetherNFT.sol](https://github.com/thousandetherhomepage/ketherhomepage/blob/master/contracts/KetherNFT.sol): Core contract is done
- [x] [KetherNFTRenderer.sol](https://github.com/thousandetherhomepage/ketherhomepage/blob/master/contracts/KetherNFTRender.sol): tokenURI on-chain rendering is functional but needs improvement.
- [x] `allAds` selector to make frontend rendering possibly with one query
- [x] Decide on `adminDisableRenderUpgrade`?
- [x] ~Rinkeby deployment~ (we have one but need to do it fresh) 
- [x] Finalize tokenURI JSON structure
  - [x] `forceNSFW` should override `nsfw` rather than be independent, right?
  - [x] NSFW state not showing up on OpenSea. Need to change type?
- [x] Confirm `owner()` function matches OpenSea ownership requirements
- [x] Confirm using the wrong `_owner` address in step 3 of minting (`wrap(...)`) will revert
- [x] Rinkeby deployment (again)
- [x] Write Etherscan instructions
  - [x] Does seed need to be kept secret?
- [x] Mainnet deployment
- [x] Minting frontend

Quick summary of how it works:
1. `Kether.precompute(uint _idx, address _owner) returns (bytes32 salt, address predictedAddress)` generates a precommitted address that will act as an atomic escrow for changing the ownership of the original ad to the NFT contract _and_ minting a corresponding NFT to your wallet. This happens client-side, so it does not cost any gas.
2. `KetherHomepage.setAdOwner(uint256 _idx_, address _newOwner)`  transfers the ownership of the ad on the original contract to the precommitment address. This is on-chain.
3. `KetherNFT.wrap(uint _idx, address _owner)` produces an ephemeral contract (`FlashEscrow`) at the precommitted address which atomically transfers the ad to the custody of the new KetherNFT contract and mints the NFT to the original owner. This is also a transaction on-chain.

To summarize the summary:  2 on-chain transactions to convert an ad to an NFT.

Anyone can "upgrade" their old ad to a new NFT ad, but all the old ads continue to be valid in non-NFT form. New ads purchased through the NFT contract are automatically reflected on both contracts-somewhat more gas-expensive, but there's no extra NFT-minting steps.
